src/Workspaces/Core/Portable/Execution/ChecksumTreeBuilder.cs

```diff
@@ -133,58 +134,96 @@ private Task<ChecksumCollection> BuildAsync(IReadOnlyList<AnalyzerReference> key
             return CreateChecksumCollectionsAsync(documentStates, kind, snapshotBuilder.BuildAsync, cancellationToken);
         }
 
-        private Task<ChecksumCollection> CreateChecksumCollectionsAsync(
+        private async Task<ChecksumCollection> CreateChecksumCollectionsAsync<TValue, TChecksumObject>(
+            IEnumerable<TValue> items, string kind, Func<TValue, CancellationToken, Task<TChecksumObject>> buildAsync, CancellationToken cancellationToken) where TChecksumObject : ChecksumObject
+        {
+            using (var pooledObject = Creator.CreateList<Task<TChecksumObject>>())
+            {
+                // create asyn checksums concurrently
+                var tasks = pooledObject.Object;
+                foreach (var item in items)
+                {
+                    tasks.Add(buildAsync(item, cancellationToken));
+                }
+
+                await Task.WhenAll(tasks).ConfigureAwait(false);
+
+                var checksums = new Checksum[tasks.Count];
+
+                for (var i = 0; i < tasks.Count; i++)
+                {
+                    var task = tasks[i];
+
+                    // task is already completed. so wait will return right away as long as there is no exception.
+                    // if there is exception, then it will throw it. if it is cancellation exception, it will make sure
+                    // it throws cancellation exception rather than aggregated exception.
+                    task.Wait(cancellationToken);
```

Comment:
> well, if await caused people to stop and trying to figure out why? then that is equally taking as much time for people to audit as using Wait.

Every time we audit we will find this code and have to re-resolve it as being ok.  I do not want to incur that tax on all future checks around async-await.  This code is not special and should not behave differently from the rest of hte IDE code around async/sync.  Please follow the patterns we have and just add a comment if you think the code requires it.  Then we won't have to continually address this code in the future.

--

Note: if the code does go through like this, my expectation is that a future checkin will simply undo the .Wait and move it a proper 'await' so that it does not get flagged in the future.  So we might as well just get the code right before that happens.